src/main/java/core/basesyntax/PurchaseOperation.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax;
+
+public class PurchaseOperation implements OperationHandler {
+    public void handle(FruitTransaction transaction) {
+        Storage.subtract(transaction.getFruit(), transaction.getQuantity());
```

Comment:
Ensure that the resulting balance in the storage is not negative after a purchase. If it is, throw a `RuntimeException` as per the checklist requirement: 'Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.'